File: csharp-reference/Characters/Piccolo.cs
using DragonBallArena.Models;

namespace DragonBallArena.Characters;

public sealed class Piccolo(string? name = null) : BaseFighter(name)
{
    public override string Id => "piccolo";
    public override string DisplayName => "Piccolo";
    public override string Greeting => "Stay focused.";
    public override string SignatureAttack => "Special Beam Cannon";
    public override string AttackStyle => "spiral";
    public override double Scale => 1.03;
    public override FighterStats Stats => new(8000, 78, 88, 0.14, 18);
    public override FighterPalette Palette => new(
        "#65b84c", "#ffffff", "#592b7d", "#f1f2ef", "#d36ab0", "#6cff74");
}

File: src/game/characters/Piccolo.ts
import { BaseFighter } from "../BaseFighter";

export class Piccolo extends BaseFighter {
  readonly id = "piccolo" as const;
  readonly name = "Piccolo";
  readonly greeting = "Stay focused.";
  readonly signatureAttack = "Special Beam Cannon";
  readonly attackStyle = "spiral" as const;
  readonly power = 8000;
  readonly speed = 78;
  readonly defense = 88;
  readonly criticalChance = 0.14;
  readonly baseDamage = 18;
  readonly nativeFacing = "right" as const;
  readonly scale = 1.03;
  readonly palette = {
    skin: "#65b84c",
    hair: "#ffffff",
    primary: "#592b7d",
    secondary: "#f1f2ef",
    accent: "#d36ab0",
    aura: "#6cff74"
  };
}
